### PR TITLE
See Fwdd shards

### PR DESCRIPTION
**Pull request by mixmix**
selection_206

selection_205

What I've done: 
- temporarily disabled the high level forwarding tab
  - because I wanted to show more detail about a particular users forwarding state, and couldn't see a tidy way to do it with that page atm
- added "tabs" to the FriendShow
  - Shards
  - **Advanced**
- Under the new advanced section put : 
  - `Forward all shards button` that changes modal to new forwarding modal
  - a history of the state of forwarding

NOTES ON CURRENT LIMITATIONS:
- the flow from opening a user to fwdding shards, so seeing the result of that is not smooth at the moment because of how modals are being usd. meh
- the query to get all fwds is a little heavy
- you can fwd the same shards multiple times
  - the counter just counts total forwards to each user, doesn't track which might be unique, or which you've sent ...
- `views/friends/show.js` is bloated:
  - e.g. queries + jumble of components in there
  - these need refactoring, but it's too soon to do it - doing it now before the designs are settled would be work which would be thrown away. Lets leave it messy ! :hankey:

### CODE REVIEW

**Comment by mixmix** [comment]
@KGibb8 this is my best guess at a balance of : 
- keeping the shards return and forwarding seperate
- being able to show details on a user all in a collected location

I'm open to putting more guard rails in place if this feels too easy to mess up. In particular I think this needs helper text about what forwarding means and what best practice is (maybe that last part in a "read more" expansion"

**Comment by ameba23** [comment]
woo looks great. just used it to forward some shards and see them afterwards. 

a question, for my own learning in what im doing.  so i noticed you are not using the same model as with `views/crystal/index` which uses `pull.drain` and then `throttle` (which isn't documented and i've forgotten what it does).  but we also don't have something like `watchForUpdates()` which `views/crystals/show` uses.  I'm guessing this is because we don't need to worry about updates because we are only waiting for messages from ourself.  is that right?

**Comment by mixmix** [comment]
re: throttle I wrote a demo here for posterity : %ACD6D8q5UgKY2YI5ejfRNOpHNMOeHNCMhtaLTtmz3+Q=.sha256 (mentioned you in thread)

we need `watchForUpdates()` for live-updating views. In the case of this view I'm not doing live updating because the we manually get the data each time the view is shown. So in our case when the forward happens the modal closes. When the person opens it again, it gets the current data which now shows the updated state.

If we wanted it to update faster we'd put a call to re-get the data in the callback to the "publish forward messages" part.

TL;DR the pattern I'm enjoying is:
- for your own actions, manually trigger re-fetch of data after we publish something (as needed)
- for others actions, if we need to see them immediately, then use a watch which triggers re-fetch

**Comment by mixmix** [comment]
@ameba23 some feedback on this review. Similar to a previous comment, I don't actually know if you've reviewed this or whether woohoo, means "yes merge this"

I'd like to talk about our PR communication sometime. e.g. we could basically just add a little vocab like : 
- I have looked at this but not reviewed the code
- I have reviewed this and it's ready to merge
- I don't know this sort of code, I'd like @ Kieran to take over this review

Given we're coming up on some 'deadlines' and you've smoke tested this (as in started it up to see if smoke pours out of the ports) and had a bit of a read of the code I'm going to merge this, so we can focus on integrating the other views.

I expect we'll be revisiting this though

**Comment by ameba23** [comment]
@mixmix 

>  woo looks great. just used it to forward some shards and see them afterwards.

woo was supposed to mean this gets my approval. probably i should have made that clearer.  i've read your code and had a play with it, but yes - i'm not super confident with this front-end stuff, and probably its not a bad idea if @KGibb8 takes another look.
